cmd/kubeadm/app/util/config/joinconfiguration.go

```diff
@@ -108,10 +107,11 @@ func JoinConfigFileAndDefaultsToInternalConfig(cfgPath string, defaultversionedc
 	if err := SetJoinDynamicDefaults(internalcfg); err != nil {
 		return nil, err
 	}
+	// TODO: this need to moved out
```

Comment:
@fabriziopandini if we have this, we cannot run `kubeadm join phase preflight` but we need always to have a token or file.
because this function is calling this
https://github.com/kubernetes/kubernetes/blob/3ec18a5aed28b46d8d275cfda2bdae2402d75215/cmd/kubeadm/app/apis/kubeadm/validation/validation.go#L112

and we are setting TOKEN = nil

Reply:
` kubeadm join phase preflight 
discovery: Invalid value: "": bootstrapToken or file must be set
`

Reply:
maybe i'm missing something here? :thinking:

Reply:
imho we were assuming the past that `kubeadm join` was always called with a bootstrap-token. should be also the preflight phase be like that? i was thinking to run without that parameter but i think this make more complicated the things. tia for hint :rose: